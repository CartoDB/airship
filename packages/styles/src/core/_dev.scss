@import 'tools/_mixins';
@import 'variables/_variables';
