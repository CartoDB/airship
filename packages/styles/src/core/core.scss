@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
