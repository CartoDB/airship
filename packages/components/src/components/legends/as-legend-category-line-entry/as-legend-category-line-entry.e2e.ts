import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-legend-category-line-entry', async () => {
  describe('Rendering', async () => {
    let page: E2EPage;

    it('should render label and color, with 1.5px solid style by default', async () => {
      page = await newE2EPage({ html: `
        <as-legend-category-line-entry>
        </as-legend-category-line-entry>
      ` });

      const element = await page.find('as-legend-category-line-entry');
      element.setProperty('label', 'Hello there');
      element.setProperty('color', '#FABADA');
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should support custom stroke width', async () => {
      page = await newE2EPage({ html: `
        <as-legend-category-line-entry>
        </as-legend-category-line-entry>
      ` });

      const element = await page.find('as-legend-category-line-entry');
      element.setProperty('label', 'Hello there');
      element.setProperty('color', '#FABADA');
      element.setProperty('width', '10');
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should support custom stroke style', async () => {
      page = await newE2EPage({ html: `
        <as-legend-category-line-entry>
        </as-legend-category-line-entry>
      ` });

      const element = await page.find('as-legend-category-line-entry');
      element.setProperty('label', 'Hello there');
      element.setProperty('color', '#FABADA');
      element.setProperty('strokeStyle', 'dotted');
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
