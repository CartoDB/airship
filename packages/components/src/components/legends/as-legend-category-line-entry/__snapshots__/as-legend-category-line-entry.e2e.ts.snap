// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-legend-category-line-entry Rendering should render label and color, with 1.5px solid style by default 1`] = `"<as-legend-category-line-entry class=\\"hydrated\\"><div class=\\"as-legend-category-line-entry--wrapper\\"><div class=\\"as-legend-category--figure--wrapper\\"><div class=\\"as-legend-category-line-entry--line\\" style=\\"border-top: 1.5px solid rgb(250, 186, 218);\\"></div></div><span class=\\"as-legend-category--label\\">Hello there</span></div></as-legend-category-line-entry>"`;

exports[`as-legend-category-line-entry Rendering should support custom stroke style 1`] = `"<as-legend-category-line-entry class=\\"hydrated\\"><div class=\\"as-legend-category-line-entry--wrapper\\"><div class=\\"as-legend-category--figure--wrapper\\"><div class=\\"as-legend-category-line-entry--line\\" style=\\"border-top: 1.5px dotted rgb(250, 186, 218);\\"></div></div><span class=\\"as-legend-category--label\\">Hello there</span></div></as-legend-category-line-entry>"`;

exports[`as-legend-category-line-entry Rendering should support custom stroke width 1`] = `"<as-legend-category-line-entry class=\\"hydrated\\"><div class=\\"as-legend-category-line-entry--wrapper\\"><div class=\\"as-legend-category--figure--wrapper\\"><div class=\\"as-legend-category-line-entry--line\\" style=\\"border-top: 10px solid rgb(250, 186, 218);\\"></div></div><span class=\\"as-legend-category--label\\">Hello there</span></div></as-legend-category-line-entry>"`;
